client/src/components/APIClient.ts

```diff
@@ -76,25 +85,32 @@ async function verifyIsJson(response) {
   throw new Error(msg);
 }
 
-function checkStatus(response) {
+function checkStatus(response: Response) {
   if (response.status >= 200 && response.status < 300) {
     return response;
   }
-  const error = new Error(`HTTP Error ${response.statusText}`);
-  error.status = response.statusText;
-  error.response = response;
+  const error = new HTTPError(response);
   console.log(error); // eslint-disable-line no-console
   throw error;
 }
 
-function parseJSON(response) {
+class HTTPError extends Error {
```

Comment:
Rather than creating a regular `Error` and then adding arbitrary properties to it, which TypeScript doesn't like, we're just going to create our own `Error` subclass instead.  This is a best practice anyways, since it allows exception handlers to check if the caught exception is an `instanceof HTTPError` and then do custom logic based on it.